components/core/src/clp/Array.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef CLP_ARRAY_HPP
+#define CLP_ARRAY_HPP
+
+#include <cstddef>
+#include <cstring>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include "ErrorCode.hpp"
+#include "TraceableException.hpp"
+
+namespace clp {
+/**
+ * Class for a run-time fix-sized array.
+ * @tparam T The type of the element in the array. The type must be default constructable.
```

Comment:
```suggestion
 * Class for a runtime fix-sized array.
 * @tparam T The type of elements in the array. The type must be default initializable so that the
 * class doesn't need to implement an initializer list constructor.
```

Is that accurate?

Reply:
yes, that's more accurate. Thanks!